File: backend-nestjs/auth-service/src/organization/organization.entity.ts
import { Entity, Column, PrimaryColumn, CreateDateColumn, UpdateDateColumn } from 'typeorm';

@Entity('organizations')
export class OrganizationEntity {
  @PrimaryColumn({ type: 'bigint' })
  id: number;

  @Column({ name: 'tenant_id', type: 'bigint' })
  tenantId: number;

  @Column({ name: 'parent_id', type: 'bigint', nullable: true })
  parentId: number | null;

  @Column({ length: 255 })
  name: string;

  @Column({ length: 50 })
  code: string;

  @Column({ length: 20, default: 'active' })
  status: string;

  @CreateDateColumn({ name: 'created_at' })
  createdAt: Date;

  @UpdateDateColumn({ name: 'updated_at' })
  updatedAt: Date;
}

File: backend-java/cdm-auth/src/main/java/com/cdm/auth/entity/OrganizationEntity.java
package com.cdm.auth.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.cdm.auth.vo.OrganizationVo;
import com.cdm.common.domain.BaseEntity;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@TableName("organizations")
@Getter @Setter @NoArgsConstructor
public class OrganizationEntity extends BaseEntity {

    private Long tenantId;
    private Long parentId;
    private String name;
    private String code;
    private String status = "active";

    public static OrganizationVo toVo(OrganizationEntity entity) {
        if (entity == null) return null;
        return OrganizationVo.builder()
                .id(entity.getId())
                .tenantId(entity.getTenantId())
                .parentId(entity.getParentId())
                .name(entity.getName())
                .code(entity.getCode())
                .status(entity.getStatus())
                .build();
    }
}
